de.marw.cdt.cmake.core/src/main/java/de/marw/cdt/cmake/core/internal/BuildscriptGenerator.java

```diff
@@ -147,13 +155,66 @@ private IFolder getBuildFolder() {
       String buildPathString = buildP.toString();
       buildPathString = mngr.resolveValue(buildPathString, "", "",
           ManagedBuildManager.getDescriptionForConfiguration(config));
-      buildP= new Path(buildPathString);
+      buildP = new Path(buildPathString);
     } catch (CdtVariableException e) {
       log.log(new Status(IStatus.ERROR, CdtPlugin.PLUGIN_ID,
           "variable expansion for build directory failed", e));
     }
 
-    buildFolder = project.getFolder(buildP);
+    // if the path is absolute, it is necessary to link the external directory
+    // into the project to use it
+    if (buildP.isAbsolute()) {
```

Comment:
Creation of the linked folder should not happen here. getBuildFolder() is called frequently and should be fast. Creation of the linked folder should happen when the user hitss apply in the project configuration page, to give immediate feed-back.
The actual build directory will happen when the projects builds.